libbeat/processors/actions/json_decode_fields.go

```diff
@@ -0,0 +1,91 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type jsonDecodeFields struct {
+	Fields []string
+}
+
+func init() {
+	processors.RegisterPlugin("json_decode_fields",
+		configChecked(newJSONDecodeFields,
+			requireFields("fields"),
+			allowedFields("fields", "when")))
+}
+
+func newJSONDecodeFields(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []string `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the json_decode_fields configuration: %s", err)
+	}
+
+	f := jsonDecodeFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f jsonDecodeFields) Run(event common.MapStr) (common.MapStr, error) {
+	for _, field := range f.Fields {
+		var output map[string]interface{}
+
+		data, err := event.GetValue(field)
+		if err != nil {
+			return event, fmt.Errorf("no %s field found: %s", field, err)
```

Comment:
I looked at the drop_field processor and looks like it takes the same approach as well. 
`err := event.Delete(field)
		if err != nil {
			return event, fmt.Errorf("Fail to delete key %s: %s", field, err)
		}`

mapstr.GetValue() returns an error when it doesn't find a key. I'm hesitant to use a hasKey + getValue since it will involve parsing the data twice. 

Any suggestions on how to fix this ?